src/dllmain.cpp

```diff
@@ -106,16 +107,104 @@ typedef struct {
 
 void Logging()
 {
-    loguru::add_file("MGSM2Fix.log", loguru::Truncate, loguru::Verbosity_MAX);
-    loguru::set_thread_name("Main");
+    // Get game name and exe path
+    char exePath[_MAX_PATH] = { 0 };
+    GetModuleFileName(gBaseModule, exePath, MAX_PATH);
+    sExePath = exePath;
+    sExeName = (sExePath / sExePath.filename()).string();
 
+    bool logDirExists = std::filesystem::is_directory(sExePath / "logs");
+    if (!logDirExists)
+    {
+        std::filesystem::create_directory(sExePath / "logs"); //create a "logs" subdirectory in the game folder to keep the main directory tidy.
+    }
+    loguru::add_file((sExePath / "logs" / "MGSM2Fix.log").string().c_str(), loguru::Truncate, loguru::Verbosity_MAX);
+    loguru::set_thread_name("Main");
+    if (!logDirExists)
+    {
+        LOG_F(INFO, "New log subdirectory created.");
+    }
     LOG_F(INFO, "MGSM2Fix v%s loaded", sFixVer.c_str());
 }
 
+///Prints CPU, GPU, and RAM info to the log to expedite common troubleshooting, since it's usually folks who are below minumum specs.
+void LogSysInfo()
+{
+#ifndef _WIN32
+    LOG_F(INFO, "System Details - Steam Deck/Linux");
+    return;
+#endif
+
+
+    std::array<int, 4> integerBuffer = {};
+    constexpr size_t sizeofIntegerBuffer = sizeof(int) * integerBuffer.size();
+    std::array<char, 64> charBuffer = {};
+    std::array<std::uint32_t, 3> functionIds = {
+        0x8000'0002, // Manufacturer  
+        0x8000'0003, // Model 
+        0x8000'0004  // Clock-speed
+    };
+
+    std::string cpu;
+    for (int id : functionIds)
+    {
+        __cpuid(integerBuffer.data(), id);
+        std::memcpy(charBuffer.data(), integerBuffer.data(), sizeofIntegerBuffer);
+        cpu += std::string(charBuffer.data());
+    }
+
+    LOG_F(INFO, "System Details - CPU: {}", cpu);
+
+    std::string deviceString;
+    for (int i = 0; ; i++)
+    {
+        DISPLAY_DEVICE dd = { sizeof(dd), 0 };
+        BOOL f = EnumDisplayDevices(NULL, i, &dd, EDD_GET_DEVICE_INTERFACE_NAME);
+        if (!f)
+        {
+            break; //that's all, folks.
+        }
+        char deviceStringBuffer[128];
+
+#ifdef UNICODE
+        WideCharToMultiByte(CP_UTF8, 0, dd.DeviceString, -1, deviceStringBuffer, sizeof(deviceStringBuffer), NULL, NULL);
+#else
+        // Convert ANSI to wide first
+        wchar_t wDeviceString[128];
+        MultiByteToWideChar(CP_ACP, 0, dd.DeviceString, -1, wDeviceString, 128);
+        WideCharToMultiByte(CP_UTF8, 0, wDeviceString, -1, deviceStringBuffer, sizeof(deviceStringBuffer), NULL, NULL);
+#endif
```

Comment:
since i'm not able to compile i can't test this. i know the unicode handling works as its what we use with MGSHDFix.